packages/google_maps_flutter/google_maps_flutter_ios/example/ios14/integration_test/google_maps_test.dart

```diff
@@ -1292,6 +1383,33 @@ void main() {
     ));
     await controllerCompleter.future;
   });
+
+  testWidgets('markerWithPinConfig', (WidgetTester tester) async {
+    final Set<AdvancedMarker> markers = <AdvancedMarker>{
+      AdvancedMarker(
+        markerId: const MarkerId('1'),
+        icon: BitmapDescriptor.pinConfig(
+          backgroundColor: Colors.green,
+          borderColor: Colors.greenAccent,
+          glyph: const TextGlyph(text: 'A', textColor: Colors.white),
+        ),
+      ),
+    };
+
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: ui.TextDirection.ltr,
+        child: ExampleGoogleMap(
+          initialCameraPosition: const CameraPosition(
+            target: LatLng(10.0, 20.0),
+          ),
+          markers: markers,
+          markerType: MarkerType.advancedMarker,
```

Comment:
You should use a completer to wait for `onMapCreated`. See test `markerWithLegacyBytes` above for an example